src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,42 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final String LINE_SEPARATOR = "-";
+    private static final String CORRECT_NATIONALITY = "Ukrainian";
+    private static final int ENOUGH_YEARS = 10;
+    private static final int MINIMAL_AGE = 35;
+    private static final int START_YEAR_INDEX = 0;
+    private static final int END_YEAR_INDEX = 1;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate != null
+                && candidate.getAge() >= MINIMAL_AGE
+                && CORRECT_NATIONALITY.equalsIgnoreCase(candidate.getNationality())
+                && candidate.isAllowedToVote()
+                && hasSufficientYearsInUkraine(candidate.getPeriodsInUkr());
+    }
+
+    private boolean hasSufficientYearsInUkraine(String periodsInUkr) {
```

Comment:
Look at an example of method
```suggestion
    private boolean checkTimeLivingInCountry(Candidate candidate) {
        String[] dates = candidate.getPeriodsInUkr().split(SEPARATOR);
        return Integer.parseInt(dates[END_INDEX])
                - Integer.parseInt(dates[BEGIN_INDEX]) >= PERIOD_IN_UKR;
    }
```

Reply:
Changed